library/kotlin/src/io/envoyproxy/envoymobile/Envoy.kt

```diff
@@ -53,4 +56,22 @@ class Envoy @JvmOverloads constructor(
   fun isTerminated(): Boolean {
     return runner.state == Thread.State.TERMINATED
   }
+
+  override fun send(request: Request, responseHandler: ResponseHandler): StreamEmitter {
+    val stream = engine.startStream(responseHandler.underlyingObserver)
+    stream.sendHeaders(request.headers, false)
+    return EnvoyStreamEmitter(stream)
+  }
+
+  override fun send(request: Request, data: ByteBuffer?, trailers: Map<String, List<String>>, responseHandler: ResponseHandler): CancelableStream {
+    val stream = engine.startStream(responseHandler.underlyingObserver)
```

Comment:
In iOS, I actually call into the other `send` function instead of duplicating some of the logic here. Not a blocker, but some nice cleanup